src/importer/DND5SpellImporter.ts

```diff
@@ -34,34 +33,26 @@ export class DND5SpellImporter {
 
     public importSpellsByUrl = async (url: string): Promise<void> => {
         try {
-            axios.get(url).then(response => {
-                const spellLocations: multiSpellImport = response.data;
-                const spells: spellImport[] = [];
-                // collect errors that occur during the import
-                const errorList: string[] =[];
-                spellLocations.results.forEach(async location => {
-                    try {
-                        axios.get(`http://dnd5eapi.co${location.url}`).then(resp => {
-                        spells.push(resp.data);
-                        // check if last spell received
-                        if ((spellLocations.results.length + errorList.length) == spells.length) {
-                            if (errorList.length) {
-                                console.warn('Errors during spell import: ',errorList);
-                            }
-                            // import all received spells
-                            this.importSpellsByData(spells);
-                        }
-                    });
-                    } catch (error) {
-                        errorList.push(`An error occured while getting ${location.url}.`);
-                    }
-                });
-            });
+            const multiSpellImport: multiSpellImport = (await axios.get(url)).data;
+            const spells = await this.getSpellImportsLocations(multiSpellImport.results);
+            this.importSpellsByData(spells);
         } catch (error) {
             console.error('While importing spells form url', url, 'an error occured.\n', error);
         }
     };
 
+    private async getSpellImportsLocations(locations: entityLocation[]): Promise<spellImport[]> {
+
+        const spells: Promise<spellImport[]> = Promise.all(locations.map(async (location) => {
+            try {
+                return (await axios.get(`http://dnd5eapi.co${location.url}`)).data;
```

Comment:
Man könnte das in einen HttpService auslagern der Requests an diese API schickt und die Base Url als konstanten Wert hat in der Zukunft :)

Reply:
Das ist derwe,den ich normalerweise in Angular gehe:)

Reply:
der Weg, nicht derwe